docs/examples/ex28.py

```diff
@@ -71,16 +72,16 @@ def make_mesh(halfheight: float,  # mm
        for label in ['heated', 'fluid-outlet', 'solid-outlet']}}
 
 
-@bilinear_form
-def conduction(u, du, v, dv, w):
-    return w.w * sum(du * dv)
+@BilinearForm
+def conduction(u, v, w):
+    return dot(w['w'] * grad(u), grad(v))
 
 
-@bilinear_form
-def advection(u, du, v, dv, w):
-    _, y = w.x
+@BilinearForm
+def advection(u, v, w):
+    y = w['x'].value[1]
```

Comment:
Can you write `_, y = w.x` ?

`w` is of type `FormDict(dict)` for which `__getattr__` has been overloaded as follows:
```python
    def __getattr__(self, attr):
        return self[attr].value
```

Reply:
Oh, yes, that's much better.  I thought that I had tried that and it hadn't worked but it does.  That's fine.  Thanks.